### ISSUE TITLE
ETDF field widget, when it allows ranges, incidentally forbids times.

### ISSUE THREAD

**Issue by rosiel** [Islandora/documentation#1364]
From the [EDTF spec](https://www.loc.gov/standards/datetime/), 

>Three conformance levels are defined: level 0, level 1, and level 2. [...] An implementation must support all of the features listed for level 0. The vendor must state one of the following levels of support:
> -  Level 0 is supported.
> -  Level 0 is supported, and in addition the following features of levels 1 and 2 are supported (list features).
> -  Level 1 is supported.
> -  Level 1 is supported, and in addition the following features of level 2 are supported (list features).
> -  Level 2 is supported.

There are some features defined within Level 0 that we do not support:

## Date and Time 
[dateI][“T”][time][shiftHour]
Date and time with timeshift in hours (only)
Example 3       ‘1985-04-12T23:20:30-04’ refers to the date 1985 April 12th time of day 23:20:30 with time shift of 4 hours behind UTC.

[oops, edited. it seems we do support timeshift when given in hours and minutes]

## Time Interval 
EDTF Level 0 adopts representations of a time interval where both the start and end are dates: start and end date only; that is, both start and duration, and duration and end, are excluded. Time of day is excluded.

- Example 1          ‘1964/2008’ is a time interval with calendar year precision, beginning sometime in 1964 and ending sometime in 2008.
- Example 2          ‘2004-06/2006-08’ is a time interval with calendar month precision, beginning sometime in June 2004 and ending sometime in August of 2006.
- Example 3          ‘2004-02-01/2005-02-08’ is a time interval with calendar day precision, beginning sometime on February 1, 2004 and ending sometime on February 8, 2005.
- Example 4          ‘2004-02-01/2005-02’ is a time interval beginning sometime on February 1, 2004 and ending sometime in February 2005. Since the start endpoint precision (day) is different than that of the end endpoint (month) the precision of the time interval at large is undefined.
- Example 5          ‘2004-02-01/2005’ is a time interval beginning sometime on February 1, 2004 and ending sometime in 2005. The start endpoint has calendar day precision and the end endpoint has calendar year precision. Similar to the previous example, the precision of the time interval at large is undefined.
- Example 6          ‘2005/2006-02’ is a time interval beginning sometime in 2005 and ending sometime in February 2006.

Until we do, we can't say we support EDTF, even though we also _do_ support some of the formats in levels 1 and 2.

I am changing the documentation for Metadata to describe the particular subset of EDTF that is implemented.

**Comment by seth-shaw-unlv** [comment]
Um, we _do_ support time interval. _However_, Islandora Defaults has date intervals turned off by default. These can easily be enabled using the Manage Form Display configuration (`/admin/structure/types/manage/islandora_object/form-display`):

<img width="1194" alt="Screen Shot 2019-11-22 at 10 41 17 AM" src="https://user-images.githubusercontent.com/29869988/69451655-c2630e00-0d14-11ea-82f7-4ee3537d8277.png">

**Comment by rosiel** [comment]
Oooh ooh ooh! Good to know! thanks. I think we're really starting to need a separate page just for EDTF, because I want to mention that as well. I'm going to go re-verify this, because that takes care of most of the things missing.

There was one more non-range limitation under Level 0 - this format:
`[dateI][“T”][time][shiftHour]`
where it's shiftHour expressed as an hour only. Am I missing anything with that?

**Comment by seth-shaw-unlv** [comment]
No, the shiftHour failure is probably a legit oversight/bug. I probably missed it in my test-suite.

**Comment by rosiel** [comment]
Hmmm. I'm testing and having good luck with ranges now. But is it expected behaviour that if the widget _can allow ranges_, it _forbids times_?

**Comment by seth-shaw-unlv** [comment]
EDTF does not allow times in ranges. **However**, the widget _should_ allow you to _either_ enter a single date with time OR use range operator without times.

If the "allow intervals" setting is enabled, you should _ be able to add a single date with time. If not, we need to file a bug report.

**Comment by kspurgin** [comment]
Meta-issue: #1748
